docs/roadmap/scaling-2019.md

```diff
@@ -0,0 +1,136 @@
+2. Track cold start latency over time by span.
+3. Performance test for cold starts.
+4. Local scheduling.
+
+**Project**: [Project 8](https://github.com/knative/serving/projects/8)
+
+### Overload Handling
+
+Knative Serving provides concurrency controls to limit the number of requests a container must handle simultaneously. Additionally, each pod has a queue for holding requests when the container concurrency limit has been reached. When the pod-level queue overflows, subsequent request are rejected with 503 "overload".
+
+This is desirable to protect the Pod from being overloaded. But the aggregate behavior is not ideal for situations when autoscaling needs some time to react to sudden increases in request load.  This could be when the revision is scaled to zero.  Or when the revision is already running some pods, but not nearly enough.
+
+The goal of Overload Handling is to enqueue requests at a revision-level. Scale-from-zero should not overload if autoscaling can react in a reasonable amount of time to provide additional pods. When new pods come online, they should be able to take load from the existing pods. Even when scaled above zero, brief spikes of overload should be handled by enqueuing requests at a revision-level. The depth of the revision-level queue should also be configurable because even the Revision as a whole needs to guard against overload.
+
+The overall problem touches on both Networking and Autoscaling, two different working groups. Much of the overload handling will be implemented in the Activator, which is a part of ingress. So this project is shared jointly between the two working groups.
+
+**Goal**: requests can be enqueued at the revision-level in response to high load.
+
+**Key Steps**:
+1. Handle overload gracefully in the Activator. Proxy requests at a rate the underlying Deployment can handle. Reject requests beyond a revision-level limit.
+2. Wire the Activator into the serving path on overload.
+3. Performance tests for overload scenarios.
+
+**Project**: [Project 7](https://github.com/knative/serving/projects/7)
+
+## Reliability
+
+### Autoscaling Availabilty
+
+Because Knative scales to zero, the autoscaling system is in the critical-path for serving requests. If the Autoscaler or Activator isn't available when an idle Revision receives a request, that request will not be served. The Activator is stateless and can be easily scaled horizontally. Any Activator Pod can proxy any request for any Revision. But the Autoscaler Pod is stateful. It maintains request statistics over a window of time. 
+
+We need a way for autoscaling to have higher availability than that of a single Pod. When an Autoscaler Pod fails, another one should take over, quickly. And the new Autoscaler Pod should make equivalent scaling decisions.
+
+**Goal**: the autoscaling system should be more highly available than a single Pod.
+
+**Key Steps**:
+1. Autoscaler replication should be configurable. This will likely require leader election.
```

Comment:
I find myself wantint to echo @markusthoemmes feedback on focusing this on the "what" vs. the "how".  Is linking to the Projects sufficient?

Reply:
Yeah.  @markusthoemmes merged a PR which makes the goals non-prescriptive.